lib/hanami/action/base_params.rb

```diff
@@ -45,7 +45,7 @@ class BaseParams
       def initialize(env)
         @env    = env
         @raw    = _extract_params
-        @params = Utils::Hash.new(@raw).symbolize!.to_h
+        @params = Utils::Hash.new(@raw).deep_dup.symbolize!.to_h
```

Comment:
I think there is a way to improve this code, since it looks quite complicated and not good in term of performance 😺